lib/swaps/Swaps.ts

```diff
@@ -1,67 +1,68 @@
+import assert from 'assert';
 
 type SwapDeal = {
   /** The role of the local node in the swap. */
   myRole: SwapDealRole;
-  /** The xud node pub key of the counterparty to this swap deal. */
-  peerPubKey: string;
+  /** The most updated deal status */
+  status: SwapDealStatus;
+  /** The most updated deal state. Works together with status. */
+  state: SwapDealState;
+  stateReason: String;
   /** Global order id in the XU network. */
   orderId: string;
-  dealId: string;
   /** The quantity of the order to execute as proposed by the taker. Negative when the taker is selling. */
+  // TODO: is it needed here? if yes, should be in satoshis
   proposedQuantity: number;
   /** The accepted quantity of the order to execute as accepted by the maker. Negative when the taker is selling. */
   quantity?: number;
-  /** The trading pair of the order. */
+  /** The trading pair of the order. The pairId together with the orderId are needed to find the deal in orderBook. */
   pairId: string;
   /** The number of satoshis (or equivalent) the taker is expecting to receive. */
-  takerAmount?: number;
+  takerAmount: number;
   /** The currency the taker is expecting to receive. */
   takerCurrency: string;
   /** Taker's lnd pubkey on the taker currency's network. */
-  takerPubKey: string;
+  takerPubKey?: string;
   /** The number of satoshis (or equivalent) the maker is expecting to receive. */
-  makerAmount?: number;
+  makerAmount: number;
   /** The currency the maker is expecting to receive. */
   makerCurrency: string;
-  /** Maker's lnd pubkey on the maker currency's network. */
-  makerPubKey?: string;
   /** The hash of the preimage. */
-  r_hash?: string;
+  r_hash: string;
   r_preimage?: string;
   createTime: number;
   executeTime?: number;
   competionTime?: number
 };
 
 interface Swaps {
+  // TODO: put swap.rejected and swap.accepted to work or delete them.
   on(event: 'swap.rejected', listener: (deal: SwapDeal) => void): this;
   on(event: 'swap.accepted', listener: (deal: SwapDeal, quantity: number) => void): this;
-  on(event: 'swap.completed', listener: (deal: SwapDeal) => void): this;
+  on(event: 'swap.paid', listener: (deal: SwapDeal) => void): this;
   on(event: 'swap.failed', listener: (deal: SwapDeal) => void): this;
   emit(event: 'swap.rejected', deal: SwapDeal): boolean;
   emit(event: 'swap.accepted', deal: SwapDeal, quantity: number): boolean;
-  emit(event: 'swap.completed', deal: SwapDeal): boolean;
+  emit(event: 'swap.paid', deal: SwapDeal): boolean;
   emit(event: 'swap.failed', deal: SwapDeal): boolean;
 }
 
 class Swaps extends EventEmitter {
-  /** A mapping containing two entries for each deal, one using dealId as a key and one using r_hash as a key. */
+  /** A map between r_hash and swap deals. */
   private deals = new Map<string, SwapDeal>();
 
   constructor(private logger: Logger, private pool: Pool, private lndBtcClient: LndClient, private lndLtcClient: LndClient) {
     super();
-
```

Comment:
I'd prefer keeping this blank line after `super()`, that's also consistent with how the spacing is in the rest of the project.